addons/mail/static/src/messaging/entity/chat_window_manager/chat_window_manager.js

```diff
@@ -320,35 +355,89 @@ function ChatWindowManagerFactory({ Entity }) {
             return visual;
         }
 
-        //----------------------------------------------------------------------
-        // Private
-        //----------------------------------------------------------------------
-
-        /**
-         * @override
-         */
-        _update(data) {
-            const {
-                /**
-                 * List of ordered chat windows (list of local ids)
-                 */
-                _ordered = this._ordered || [],
-                isHiddenMenuOpen = this.isHiddenMenuOpen || false,
-            } = data;
-
-            Object.assign(this, {
-                _ordered,
-                isHiddenMenuOpen,
-            });
-        }
-
     }
 
+    ChatWindowManager.entityName = 'ChatWindowManager';
+
     ChatWindowManager.fields = {
+        /**
+         * List of ordered chat windows (list of local ids)
+         */
+        _ordered: attr({ default: [] }),
+        // FIXME: dependent on implementation that uses arbitrary order in relations!!
+        allOrdered: one2many('ChatWindow', {
+            compute: '_computeAllOrdered',
+            dependencies: [
+                '_ordered',
+                'chatWindows',
+            ],
+        }),
+        allOrderedHidden: one2many('ChatWindow', {
+            compute: '_computeAllOrderedHidden',
+            dependencies: ['visual'],
+        }),
+        allOrderedHiddenThread: one2many('Thread', {
+            related: 'allOrderedHidden.thread',
+        }),
+        allOrderedHiddenThreadMessageUnreadCounter: attr({
+            related: 'allOrderedHiddenThread.message_unread_counter',
```

Comment:
How does that work, a related on a o2m?

Reply:
it acts as a `.map()` on each entity field in the relation